execution/executor/src/db_bootstrapper.rs

```diff
@@ -22,17 +28,29 @@ pub fn maybe_bootstrap_db<V: VMExecutor>(db: DbReaderWriter, config: &NodeConfig
         .execution
         .genesis
         .as_ref()
-        .expect("failed to load genesis transaction!")
+        .expect("failed to load genesis txn!")
         .clone();
 
+    execute_and_commit::<V>(db, genesis_txn).map(|_| ())
+}
+
+pub fn compute_genesis_waypoint<V: VMExecutor>(txn: Transaction) -> Result<Waypoint> {
+    let path = TempPath::new();
+    path.create_as_dir().expect("Unable to create directory");
+    let (_, db) = DbReaderWriter::wrap(LibraDB::new(&path.path()));
+    let ledger_info = execute_and_commit::<V>(db, txn)?;
+    Waypoint::new(&ledger_info)
+}
+
+fn execute_and_commit<V: VMExecutor>(db: DbReaderWriter, txn: Transaction) -> Result<LedgerInfo> {
```

Comment:
The parameter being `txn` suggests it's generic but the comments in the code suggests we expect the genesis txn. A bit weird.

Reply:
we can assert it's WaypointWriteset here

Reply:
Going to go with a middle choice of just genesis. That way when we get to WaypointWriteset, we are golden.

With respect to WaypointWriteset, I propose another PR that cleans up the genesis-builder -> config that replaces the genesis transaction with a genesis WaypointWriteset.